hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/MetricsRegistryImpl.java

```diff
@@ -145,25 +161,59 @@ public void deregisterDynamicMetricsProvider(DynamicMetricsProvider metricsProvi
+                                        ProbeFunction function) {
+        registerStaticProbeWithUnit(source, tagger, name, level, unit, function);
+    }
+
     @Override
     public <S> void registerStaticProbe(S source, String name, ProbeLevel level, LongProbeFunction<S> function) {
-        checkNotNull(source, "source can't be null");
-        checkNotNull(name, "name can't be null");
-        checkNotNull(function, "function can't be null");
-        checkNotNull(level, "level can't be null");
+        registerStaticProbeWithoutUnit(source, name, level, function);
+    }
 
-        registerInternal(source, newMetricTagger().withMetricTag(name), level, function);
+    @Override
+    public <S> void registerStaticProbe(S source, String name, ProbeLevel level, ProbeUnit unit, LongProbeFunction<S> function) {
+        registerStaticProbeWithUnit(source, newMetricTagger(), name, level, unit, function);
+    }
+
+    @Override
+    public <S> void registerStaticProbe(S source, MetricTagger tagger, String name, ProbeLevel level, ProbeUnit unit,
+                                        LongProbeFunction<S> function) {
+        registerStaticProbeWithUnit(source, tagger, name, level, unit, function);
     }
 
     @Override
     public <S> void registerStaticProbe(S source, String name, ProbeLevel level, DoubleProbeFunction<S> function) {
+        registerStaticProbeWithoutUnit(source, name, level, function);
+    }
+
+    @Override
+    public <S> void registerStaticProbe(S source, String name, ProbeLevel level, ProbeUnit unit,
+                                        DoubleProbeFunction<S> function) {
+        registerStaticProbeWithUnit(source, newMetricTagger(), name, level, unit, function);
+    }
+
+    @Override
+    public <S> void registerStaticProbe(S source, MetricTagger tagger, String name, ProbeLevel level, ProbeUnit unit,
+                                        DoubleProbeFunction<S> function) {
+        registerStaticProbeWithUnit(source, tagger, name, level, unit, function);
+    }
+
+    private <S> void registerStaticProbeWithoutUnit(S source, String name, ProbeLevel level, ProbeFunction function) {
         checkNotNull(source, "source can't be null");
         checkNotNull(name, "name can't be null");
         checkNotNull(function, "function can't be null");
         checkNotNull(level, "level can't be null");
 
-        MetricTagger metricTagger = newMetricTagger();
-        registerInternal(source, metricTagger.withMetricTag(name), level, function);
+        registerInternal(source, newMetricTagger().withMetricTag(name), level, function);
+    }
+
+    private <S> void registerStaticProbeWithUnit(S source, MetricTagger tagger, String name, ProbeLevel level, ProbeUnit unit,
+                                                 ProbeFunction function) {
+        registerInternal(source, tagger.withTag("unit", unit.name().toLowerCase())
```

Comment:
Shouldn't this method include null checks similar to `#registerStaticProbeWithoutUnit`?